keras/utils/vis_utils.py

```diff
@@ -13,7 +13,13 @@
 
 
 def _check_pydot():
-    if not (pydot and pydot.find_graphviz()):
+    try:
+        if hasattr(pydot, 'find_graphviz'):
+            if not pydot.find_graphviz():
+                raise ImportError()
+        else:
+            pydot.Dot.create(pydot.Dot())
+    except:
```

Comment:
The problem with this is that pydot raises a generic exception if it can't find dot, so there is no specific class. One alternative might be to not checking the dot installation at all, but that produces a less helpful error message that doesn't mention graphviz specifically.